### ISSUE TITLE
Testsuite failures on wildly

### ISSUE THREAD

**Issue by gtudan** [eclipse-ee4j/krazo#70]
Our test suite has a couple of tests that fail on wildly:

- [ ] The Asciidoctor-ext has class loading issues on wildly. JRuby doesn't find the Asciidoctor module, which is a [known issue](https://github.com/asciidoctor/asciidoctorj/blob/26641dbe1510a83fbb9c5cab3974a8736b385bdd/README.adoc#running-asciidoctorj-on-wildfly-as) at Asciidoctorj
- [ ] JSR-223 fails for python. Probably something similar as in the jruby/asciidoctor issue above. 
- [x] The Jetbrick test fails and I'm not sure why. The project documentation and issue tracker is in mandarin(?), so looking for known issues is a bit difficult.

Most of those failures are probably issues with the test suite and not bugs in the extensions. But they point out potential pitfalls that should be documented somewhere (readme of the extension?).

**Comment by gtudan** [comment]
Strange - the jetbrick test works locally. Might be something on Travis.

The Jython-issue is a little odd as well. The engine get's detected by the ScriptManager, but isn't registered correctly: the SPI get's picked up, but there are no bindings for the names or extensions. The initialization is run (I can see some log output), but there's no error explaining the issue.

**Comment by gtudan** [comment]
I'm afraid getting the asciidoctorj-example working in the current test suite would be very hard. We would need to patch wildfly to add asciidoctor as module: https://github.com/asciidoctor/asciidoctorj/blob/26641dbe1510a83fbb9c5cab3974a8736b385bdd/README.adoc#running-asciidoctorj-on-wildfly-as

The guys at asciidoctor did it some time ago with a dedicated docker image: https://asciidoctor.org/news/2015/03/04/docker-asciidoctorj-announcement/
Today they use Gradle to patch wildfly: https://github.com/asciidoctor/asciidoctorj/tree/master/asciidoctorj-wildfly-integration-test

I'm not sure if it is worth the effort to build something like this for krazo. Should we exclude the Asciidoctor-Test from the wildfly-testrun?

**Comment by chkal** [comment]
> I'm not sure if it is worth the effort to build something like this for krazo. Should we exclude the Asciidoctor-Test from the wildfly-testrun?

Hmmm. Too bad. I agree that it would be fine to skip the AsciiDoc test in this case. A few years ago I did something similar in the rewrite project. 

See:

https://github.com/ocpsoft/rewrite/blob/eded0bad0101d119a3a1e67ed3f97a3bc9a8222e/impl-servlet-tests/src/test/java/org/ocpsoft/rewrite/servlet/config/ConfigurationDeploymentTest.java#L39

https://github.com/ocpsoft/rewrite/blob/eded0bad0101d119a3a1e67ed3f97a3bc9a8222e/pom.xml#L754-L762

**Comment by erdlet** [comment]
> The Jython-issue is a little odd as well. The engine get's detected by the ScriptManager, but isn't registered correctly: the SPI get's picked up, but there are no bindings for the names or extensions. The initialization is run (I can see some log output), but there's no error explaining the issue.

Hi all, 

I think I found the error (after hard debugging, because someone thinks it's a great idea to disable stacktraces with `private static final` variables). It seems, that something during the initialization of Jython doesn't work during module resolution:

```
Cannot import site module and its dependencies: No module named site
Determine if the following attributes are correct:
  * sys.path: ['/content/b8d5ff96-df15-4bc7-a604-d3696e2eb98d.war/WEB-INF/lib/Lib', '__classpath__', '__pyclasspath__/']
    This attribute might be including the wrong directories, such as from CPython
  * sys.prefix: /content/b8d5ff96-df15-4bc7-a604-d3696e2eb98d.war/WEB-INF/lib
    This attribute is set by the system property python.home, although it can
    be often automatically determined by the location of the Jython jar file

You can use the -S option or python.import.site=false to not import the site module
```

Unfortunately I'm not deep enough into Jython to fix it. And honestly, I'm not sure if its worth to fix it, as Python 2 is EOL in five months and there is no newer Jython version available.

**Comment by gtudan** [comment]
Nice - I didn't mange to get this debugged that deep. 👍 

This might be fixable by setting `python.home` or adding jboss-vfs  - but I don't think it's worth the effort. It doesn't look like there will by a jython3 anytime soon, so we should disable this test.

**Comment by erdlet** [comment]
Maybe we should even delete the Jython Test completely, as the used Python version gets outdated in five months at the latest.

But for the moment, I can add the `IgnoreOnWildfly` Annotation and update the PR with the AsciiDoc test.

**Comment by gtudan** [comment]
Closes by #88 and #102
